### PR TITLE
Improve integration test readability

### PR DESCRIPTION
**Pull request by rfeistenauer**
## Description

Improves the readability of the integrationtests.

### Things to add, update or check by the maintainers before this PR can be merged.

- [x] ~Public api function documentation~
- [x] ~Architecture documentation reflects actual code structure~
- [x] ~[Tutorials](https://wiki.ros.org/pilz_robots/Tutorials/)~
- [x] ~Overview on [ROS wiki](https://wiki.ros.org/pilz_robots)~
- [x] ~Package Readme ([example pilz_robots](https://github.com/PilzDE/pilz_robots/blob/melodic-devel/README.md))~
- [x] Good commit messages ([some tips](https://dev.to/jacobherrington/how-to-write-useful-commit-messages-my-commit-message-template-20n9))
- [x] CHANGELOG.rst updated
- [x] ~Copyright headers~
- [x] ~Examples~

### Review Checklist
- [x] ~Soft- and hardware architecture (diagrams and description)~
- [x] Test review (test plan and individual test cases)
- [x] ~Documentation describes purpose of file(s) and responsibilities~
- [x] Code (coding rules, style guide)

### Release planning (please answer)
- [x] ~When is the new feature released?~
- [x] ~Which dependent packages do have to be released simultaneously?~

### Hardware tests
_Unstrike the text below to enable automatic hardware tests if available. Otherwise use this as a request for the reviewer._
- [x] ~Perform hardware tests~

### CODE REVIEW

File: standalone/test/integration_tests/api/integrationtest_scanner_api.cpp [hunk 1]
```diff
@@ -172,19 +178,15 @@ TEST_F(ScannerAPITests, shouldReceiveStartRequestWithCorrectHostIpWhenUsingAutoI
...
-
-  util::Barrier start_req_received_barrier;
-  EXPECT_CALL(*nice_scanner_mock_, receiveControlMsg(_, data_conversion_layer::start_request::serialize(start_req)))
-      .WillOnce(OpenBarrier(&start_req_received_barrier));
+  setUpStrictScannerMock();
+  const auto start_req_received_barrier =
+      prepareStrictMockStartRequestBarrier(generateScannerConfig(HOST_IP_ADDRESS, FRAGMENTED_SCAN));
 
-  nice_scanner_mock_->startListeningForControlMsg();
+  strict_scanner_mock_->startListeningForControlMsg();
```
Comment (agutenkunst) [inline standalone/test/integration_tests/api/integrationtest_scanner_api.cpp:185]:
```suggestion
  scanner_mock_->startListeningForControlMsg();
```

File: standalone/test/integration_tests/api/integrationtest_scanner_api.cpp [hunk 2]
```diff
@@ -74,6 +74,7 @@ class ScannerAPITests : public testing::Test
   void setUpNiceScannerMock();
   void setUpStrictScannerMock();
   void prepareScannerMockStartReply();
+  std::shared_ptr<util::Barrier> prepareStrictMockStartRequestBarrier(const ScannerConfiguration& config);
```
Comment (agutenkunst) [inline standalone/test/integration_tests/api/integrationtest_scanner_api.cpp:77]: Could be `std::unique_ptr`?

Comment (rfeistenauer) [reply standalone/test/integration_tests/api/integrationtest_scanner_api.cpp]: jop

**Review by agutenkunst — APPROVED** [review]
Thank you!

**Pull request merged**